### PR TITLE
kubeadm: upgrade node should skip upgrade-addons if not control plane

### LINKED ISSUE #3109: [CI Failing] cannot find kube-apiserver image for current control plane instance kinder-super-admin-worker-1
https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-super-admin-latest
and
https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm#kubeadm-kinder-upgrade-1-31-latest

Since https://github.com/kubernetes/kubernetes/compare/abc056843...d91391451

```
I0910 10:14:36.570290    1273 round_trippers.go:560] GET https://172.17.0.2:6443/api/v1/namespaces/kube-system/pods?labelSelector=component%3Dkube-apiserver 200 OK in 4 milliseconds
cannot find kube-apiserver image for current control plane instance kinder-super-admin-worker-1
k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade.UnupgradedControlPlaneInstances
	k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade/postupgrade.go:80
k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/upgrade.shouldUpgradeAddons
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/upgrade/addons.go:65
k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/upgrade.runCoreDNSAddon
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/upgrade/addons.go:91
k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow.(*Runner).Run.func1
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow/runner.go:259
k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow.(*Runner).visitAll
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow/runner.go:446
k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow.(*Runner).Run
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow/runner.go:232
k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade.newCmdNode.func1
	k8s.io/kubernetes/cmd/kubeadm/app/cmd/upgrade/node.go:87
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.8.1/command.go:985
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.8.1/command.go:1117
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.8.1/command.go:1041
k8s.io/kubernetes/cmd/kubeadm/app.Run
	k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:47
main.main
	k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:25
runtime.main
	runtime/proc.go:272
runtime.goexit
	runtime/asm_amd64.s:1700
```

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

Fix a CI failure that is caused by https://github.com/kubernetes/kubernetes/pull/127242.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubeadm/issues/3109

#### Special notes for your reviewer:
/cc @SataQiu @neolit123 
- shouldUpgradeAddons will be called in `runCoreDNSAddon` and `runKubeProxyAddon`

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: cmd/kubeadm/app/cmd/phases/upgrade/addons.go [hunk 1]
```diff
@@ -62,6 +67,21 @@ func NewAddonPhase() workflow.Phase {
 }
 
 func shouldUpgradeAddons(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, out io.Writer) (bool, error) {
+	nodeSelector := labels.SelectorFromSet(labels.Set(map[string]string{
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/upgrade/addons.go:70]: in other places we check if /etc/kubernetes/manifests/kube-apiserver.yaml exists to say this is a cp node

Comment (SataQiu) [reply cmd/kubeadm/app/cmd/phases/upgrade/addons.go:70]: We should implement `IsControlPlaneNode() bool` for both `kubeadm upgrade apply` and `kubeadm upgrade node`

https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/phases/upgrade/node/data.go#L29

https://github.com/kubernetes/kubernetes/blob/f836773540cc83488250a12637e7d924078ef284/cmd/kubeadm/app/cmd/upgrade/node.go#L150-L154

Comment (SataQiu) [reply cmd/kubeadm/app/cmd/phases/upgrade/addons.go:70]: Then we can use `data.IsControlPlaneNode()`

just like :
```go
		// If this is not a control-plane node, this phase should not be executed.
		if !data.IsControlPlaneNode() {
			fmt.Println("[upgrade/kubeconfig] Skipping phase. Not a control plane node.")
			return nil
		}
```

Comment (SataQiu) [reply cmd/kubeadm/app/cmd/phases/upgrade/addons.go:70]: I can help send a fix if needed.

Comment (pacoxu) [reply cmd/kubeadm/app/cmd/phases/upgrade/addons.go:70]: > Then we can use `data.IsControlPlaneNode()`

IIRC, the node data doesn't implement this.


> I can help send a fix if needed.

OK.  Then I would not update this PR.

Comment (SataQiu) [reply cmd/kubeadm/app/cmd/phases/upgrade/addons.go:70]: https://github.com/kubernetes/kubernetes/pull/127289

**Comment by pacoxu** [comment]
/close
in favor of https://github.com/kubernetes/kubernetes/pull/127289

**Pull request closed without merging**